src/sql/src/catalog.rs

```diff
@@ -421,18 +394,18 @@ pub enum CatalogType<T: TypeReference> {
     Interval,
     Jsonb,
     List {
-        element_reference: T::Reference,
+        element_reference: GlobalId,
     },
     Map {
-        key_reference: T::Reference,
-        value_reference: T::Reference,
+        key_reference: GlobalId,
+        value_reference: GlobalId,
```

Comment:
These used to be called `key_id` and `value_id`, might make sense to rename them back.